### DISCUSSION TITLE
ROS2 node and Zenoh native applications

### DISCUSSION THREAD

**Discussion by uupks** [eclipse-zenoh/roadmap#110] [Q&A] [answered]
How ros2 node communicate with Zenoh native applications ？

Does zenoh native application need the ros2 message definition ？

**Comment by evshary** [comment, marked as answer]
The [blog](https://zenoh.io/blog/2021-04-28-ros2-integration/) might be what you want.
The message from ROS 2 node will be transformed into Zenoh with zenoh-bridge-dds (or zenoh-bridge-ros2dds), and then Zenoh native application can process it.
Of course, Zenoh native application should know the definition of ROS 2 message. You should have it in your program.
Here is the example: https://github.com/eclipse-zenoh/zenoh-demos/blob/master/ROS2/zenoh-python-teleop/ros2-teleop.py#L31
